_drafts/javascript.md

```diff
@@ -0,0 +1,110 @@
+---
+state: In Progress
+---
+
+# JavaScript
+
+## Motivation
+
+This evolution proposal is part of point 3 of the Hubot 3.0 milestone:
+
+> Modernize the project by translating CoffeeScript to JavaScript, improving integration with various developer tools, and adding features that make it easier for developers to automate their workflow.
+
+**Scope:**
+
+- Simplify contribution / maintenance
+- Removal of tooling / build steps
+- Align Node support with [Node’s LTS schedule](https://github.com/nodejs/LTS#lts-schedule1)
+- maintain backwards compatibility of user scripts written in CoffeeScript
+
+**Out of scope:**
+
+- no new features or breaking changes (beyond dropping support for Node <v4)
+- Introduce any other tooling besides linter
+- Any non-trivial refactoring of the existing code base
+
+## Proposed solution
+
+Converting of all CoffeeScript to JavaScript which can run natively on Node 4 ([current maintenance version until April 2018](https://github.com/nodejs/LTS#lts-schedule1)).
+
+The only tool I would like to introduce is a linting tool which will be run as part of `npm test`.
+
+I suggest [standard](https://standardjs.com/). We’ve been using it in all our projects at [Hoodie](http://hood.ie/) and [Neighbourhoodie](http://neighbourhood.ie) since 2+ years and never looked back. It's a zero-configuration JavaScript linter.
+
+## Detailed process
+
+**TODO:** make one or two sample CoffeeScript to JavaScript conversions (I’m working on it right now and will make it part of the proposal PR for the evolution repository)
+
+The steps of the conversation will be
+
+- [ ] Convert source files from CoffeeScript to JavaScript.
+- [ ] Convert test files from CoffeeScript to JavaScript.
+- [ ] update package.json
+- [ ] add script for JavaScript linting
+- [ ] Update documentation
```

Comment:
Everything in https://github.com/github/hubot/issues/1327 should also be in this list, especially https://github.com/github/generator-hubot